cluster/src/main/scala/org/apache/spark/sql/streaming/SnappyStreamingContextFactory.scala

```diff
@@ -61,13 +56,18 @@ class SnappyStreamingContextFactory extends SparkContextFactory {
     new SnappyStreamingContext(LeadImpl.getInitializingSparkContext,
       Milliseconds(interval)) with ContextLike {
 
+      private val addedJars = scala.collection.mutable.ArrayBuffer.empty[String]
```

Comment:
Hmm. You mean to see two threads trying submitting two streaming jobs at a time. Yes , probably we should guard that condition.
addJobJar will be getting called from JobServer. Please see the related PR  https://github.com/SnappyDataInc/spark-jobserver/pull/10